src/sentry/static/sentry/app/views/settings/components/forms/form.tsx

```diff
@@ -5,10 +5,52 @@ import styled from 'react-emotion';
 
 import {t} from 'app/locale';
 import Button from 'app/components/button';
-import FormModel from 'app/views/settings/components/forms/model';
+import FormModel, {
+  FormOptions,
+  FieldValue,
+} from 'app/views/settings/components/forms/model';
 import Panel from 'app/components/panels/panel';
 
-export default class Form extends React.Component {
+type Data = {[key: string]: FieldValue};
+
+type Props = {
+  apiMethod: string;
+  apiEndpoint: string;
+  children: React.ReactNode;
+
+  className?: string;
+  cancelLabel?: string;
+  submitDisabled?: boolean;
+  submitLabel?: string;
+  submitPriority?: Button['props']['priority'];
+  footerClass?: string;
+  footerStyle?: object;
+  extraButton?: React.ReactNode;
+  initialData?: {};
```

Comment:
Should this be type `Data`?

Reply:
Hmm, `Data` might just need to be `{}` for now until we want to introduce generics

Reply:
🙏 generics

Reply:
We can add generics later down the road, I think this is fine for now